app/controllers/IdentityController.scala

```diff
@@ -52,6 +53,26 @@ class IdentityController(
         }
       )
   }
+
+  def getUserType(maybeEmail: Option[String]): Action[AnyContent] = PrivateAction.async { implicit request =>
+    maybeEmail.fold {
+      SafeLogger.error(scrub"No email provided")
+      Future.successful(InternalServerError.asInstanceOf[Result])
+    } { email =>
+      identityService
+        .getUserType(email)
+        .fold(
+          err => {
+            SafeLogger.error(scrub"Failed to retrieve user type for ${email}: ${err.toString}")
+            InternalServerError
```

Comment:
I think if you make both of these `InternalServerError("whatever message")` then you don't need the `.asInstanceOf[Result]`